src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -47,14 +50,65 @@ object RockSlicer {
       Block(center, block.nonRedundantFaces)
     }
 
-    // Calculate centroid of each block
-    val centroidBlocks = nonRedundantBlocks.map { block =>
+    // Find all blocks that contain processor joints
+    val processorBlocks = nonRedundantBlocks.filter { block => 
+      block.faces.exists { face => face.processorJoint}
+    }
+
+    // Find blocks that do not contain processor joints
+    val realBlocks = nonRedundantBlocks.filter { block =>
+      val faceFlags = block.faces map { face =>
+        face.processorJoint
+      }
+      !faceFlags.contains(true)
+    }
```

Comment:
Aha, I see. I was just reading that if we want to filter the same RDD twice, we may want to take advantage of some of Spark's caching features. Another thing for us to look into I guess.